lms/djangoapps/courseware/date_summary.py

```diff
@@ -287,6 +290,14 @@ def description(self):
 
     @property
     def date(self):
+        if self.course.self_paced:
+            weeks_to_complete = get_course_run_details(self.course.id, ['weeks_to_complete']).get('weeks_to_complete')
+            if weeks_to_complete:
+                course_duration = datetime.timedelta(weeks=weeks_to_complete)
+                if self.course.end < (self.current_time + course_duration):
+                    return self.course.end
+                return None
```

Comment:
I'm about to extract this into its own helper method in the code I'm writing. We should just make it consistent when that merges.

Reply:
Sounds good. Let me know when that merges and I can update this.